packages/next/client/link.tsx

```diff
@@ -265,14 +308,14 @@ function Link(props: React.PropsWithChildren<LinkProps>) {
       }
 
       setIntersectionRef(el)
-      if (childRef) {
+      if (oldBehavior && childRef) {
```

Comment:
Looks like `childRef` is already conditional based on `oldBehavior` so we could omit this check which allows removing it from the dependency array of this `useCallback`

Reply:
I kept the check here to make it easier to clean up all cases of the old behavior when removing the feature later on. Don't feel strongly about it though.